File: NetBeans/src/edu/hdsb/gwss/nathan/ics3u/u6/Velocity.java
/*
Name:Nathan Wu
Date:5/8/2019
Version:1.0
Description:
    This program calculates you speed in meters per second with you distance and
    time inputed.
 */
package edu.hdsb.gwss.nathan.ics3u.u6;

import java.text.NumberFormat;
import java.util.Scanner;

/**
 *
 * @author 1wunat2
 */
public class Velocity {

    /**
     * @param args the command line arguments
     */
    
    //variables
    public static double distance, time, speed;
    
    public static void main(String[] args) {
        splashscreen();
        data();
        calculateSpeed();
        //output
        System.out.println("Your speed is " + speed + " meters/second");
    }
    
    public static void splashscreen(){
        System.out.println("SPEED");
        System.out.println("----------");
        System.out.println("This program will calculate you speed in m/s using"
                + "your inputed distance and time.");
    }
    
    public static void data(){
        //object
        Scanner input = new Scanner( System.in );
        
        //input
        System.out.println("\nWhat is your distance in meters?");
        distance = input.nextDouble();
        System.out.println("What is your time in seconds?");
        time = input.nextDouble();
    }
    
    public static void calculateSpeed(){
        //calculate speed
        NumberFormat decimal = NumberFormat.getNumberInstance(); 
        decimal.setMinimumFractionDigits( 2 );
        decimal.setMaximumFractionDigits( 2 );
        speed = distance / time;
    }
}

File: independant_study/Velocity.py
# imports
import math

# splash screen
def splashScreen():
    print("Speed")
    print("---------")
    print("This program will "
          "calculate your velocity in m/s using your imputed distance at time. "
          "The movement is on a plane, there is only fwd, positive distance and "
          "bwd, negative distance")


# get user input
def data():
    d = float(input("What is your distance in meters: "))
    t = float(input("What if your time in seconds: "))
    return d, t


# calculate
def calculate_velocity(d, t):
    return d * t

# determine direction
def get_direction(v):
    if v < 0:
        return "bwd"
    elif v > 0:
        return "fwd"
    return ""

splashScreen()
(distance, time) = (4, 2)
# calculate speed
speed = calculate_velocity(distance, time)
# determine direction and add it as a string to speed
velocity = str(math.fabs(speed)) + " " + get_direction(speed)
# print results
print(velocity)
